src/Moryx.AbstractionLayer.Resources.Endpoints/ResourceManagementController.cs

```diff
@@ -142,30 +143,48 @@ public ActionResult<Entry> InvokeMethod(long id, string method, Entry parameters
         public ActionResult<ResourceModel> ConstructWithParameters(string type, string method = null, [FromBody(EmptyBodyBehavior = EmptyBodyBehavior.Allow)] Entry arguments = null)
         {
             var trustedType = WebUtility.HtmlEncode(type);
-
-            return method == null ? Construct(trustedType, null)
-                : Construct(trustedType, new MethodEntry { Name = method, Parameters = arguments });
+            if (method is null)
+                return Construct(trustedType);
+            else
+                return Construct(trustedType, new MethodEntry { Name = method, Parameters = arguments });
         }
 
-        private ActionResult<ResourceModel> Construct(string type, MethodEntry method)
+        private ActionResult<ResourceModel> Construct(string type)
         {
-            var resource = (Resource)Activator.CreateInstance(_resourceTypeTree[type].ResourceType);
-            if (resource is null)
+            Resource resource;
+            try
+            {
+                resource = (Resource)Activator.CreateInstance(_resourceTypeTree[type].ResourceType);
+            }
+            catch (Exception)
```

Comment:
Apart from the fact, that I wonder why NotFound is returned for something that presumably did not exist (i.e. wasn't asked for), and what I wouldn't change within this PR: Wouldn't it be better to distinguish errors here and return a 500 or better reason for everything that does not fall into the 'not found' category?

Reply:
Yes that would definitively be better, I just saw the if-null statement and thought, well this will never happen because a cast of null will through an exception, but then I concentrated on the other construct method 🙈